ql/src/java/org/apache/hadoop/hive/ql/parse/UpdateDeleteSemanticAnalyzer.java

```diff
@@ -285,6 +296,61 @@ private void reparseAndSuperAnalyze(ASTNode tree, Table mTable, ASTNode tabNameN
     }
   }
 
+  private boolean tryMetadataUpdate(ASTNode tree, Table table, ASTNode tabNameNode) throws SemanticException {
+    TableName tableName = getQualifiedTableName(tabNameNode);
```

Comment:
could we please  introduce the feature flag so we could disable this functionality if something goes wrong

Reply:
@deniskuzZ 
Can I use this config as the feature flag -
```
HIVE_OPTIMIZE_REPLACE_DELETE_WITH_TRUNCATE("hive.optimize.delete.all", false, 
    "Optimize delete the entire data from table, use truncate instead"),
```
The purpose of this flag was to optimize the delete using truncate but I think the config itself - hive.optimize.delete.all is generic.
`HIVE_OPTIMIZE_REPLACE_DELETE_WITH_TRUNCATE`
This needs a name change. Any suggestions for the name?

Reply:
discussed in private, let's create new config for that `hive.optimize.delete.metadata.only`

Reply:
Done.